event-based-simulators/main/scripts/ExportOrImportProfileData.py

```diff
@@ -0,0 +1,144 @@
+        count += 1
+        appendDataToJsonFile(alarm.to_json(), filePath, count)
+
+
+def listMeasurements(c8y, childDevice, createFrom, createTo, filePath):
+    # Create a count variable as a json/dict key to save json data
+    count = 0
+    for measurement in c8y.measurements.select(source=childDevice.id, after=createFrom, before=createTo):
+        print(f"Found measurement id #{measurement.id}\n, type: {measurement.type}\n")
+        count += 1
+        appendDataToJsonFile(measurement.to_json(), filePath, count)
+        # for measurementKey in measurement.fragments:
+        #    print(f"{measurementKey}: {measurement.fragments.get(measurementKey)}")
+
+
+def appendDataToJsonFile(jsonData, filePath, count, json_data={}):
+    try:
+        # Load content of existing json data file
+        with open(filePath, 'r') as f:
+            json_data = json.load(f)
+            print(json_data)
+    except:
+        print(f"Create new data json file {filePath}")
+
+    # Create new json file or add data to an existing json file
+    with open(filePath, 'w') as f:
+        json_data[f"{count}"] = jsonData
+        json.dump(json_data, f, indent=2)
+        print("New data is added to file")
+
+
+def createFilePathFromDateTime(DATA_TYPE):
+    # Check if folder containing data files exists and make one if not
+    if not os.path.exists('export_data'):
+        os.makedirs('export_data')
+    # dd/mm/YY H:M:S
+    dateTimeString = datetime.now().strftime(f"{DATA_TYPE}_%d_%m_%Y_%H_%M_%S")
+    relativeFilePath = f'export_data\{dateTimeString}.json'
+    filePath = os.path.join(os.path.dirname(__file__), relativeFilePath)
+    return filePath
+
+
+def chooseOneOfChoices(listOfChoices):
+    while True:
+        listToStringWithCommas = "', '".join(listOfChoices)
+        print(f"Type one of these words '{listToStringWithCommas}'")
+        inputString = input()
+        if inputString in listOfChoices:
+            break
+        else:
+            print(f"the choice {inputString} is not accepted, only input {listToStringWithCommas}")
+    return inputString
+
+
+def checkFileList():
+    if not os.path.exists('export_data'):
+        print("No folder with name export_data")
+    else:
+        onlyfiles = [f for f in os.listdir('export_data') if isfile(join('export_data', f))]
+        return onlyfiles
+
+
+# Main function to run the script
+if __name__ == '__main__':
+    c8y = ArgumentsAndCredentialsHandler.c8yPlatformConnection()
+    # Provide input to choose the mode and data type to export
+    print("Do you want to 'import' or 'export' data?")
+    ACTION = chooseOneOfChoices(['import', 'export'])
+    print(f"You chose to {ACTION} data")
+    print("Do you want to extract 'all' devices data or only data from a 'specific' device?")
+    MODE = chooseOneOfChoices(['all', 'specific'])
+    print(f"You chose to {ACTION} data from {MODE.upper()} device(s)")
+    print("Do you want to extract 'measurements' or 'alarms' data?")
```

Comment:
could you add the option to also export both?

Reply:
- I'd suggest to use this file name pattern: "{EXTERNAL_ID}_{DATA_TYPE}_%d%m_%Y_%H_%M_%S" (i.e. sim001_profile-alarms_07_11_2022_13_49_42.json)
-- this way we can use the filename to determine the correct device/profile when importing
-- the external ID for profiles is implemented by @pejo-sag and will be merged soon; see https://github.com/SoftwareAG/oee-simulators/pull/25 https://github.com/SoftwareAG/oee-simulators/blob/13a4e6f689746377ff487bec0ba8476fba2b2679/event-based-simulators/main/oeeAPI.py#L64